src/Admin/ValidationCounts.php

```diff
@@ -89,8 +89,9 @@ public function register() {
 	 * Enqueue admin assets.
 	 */
 	public function enqueue_scripts() {
-		/** This action is documented in includes/class-amp-theme-support.php */
-		do_action( 'amp_register_polyfills' );
+		if ( $this->is_options_menu() ) {
```

Comment:
Good idea.

Reply:
Oh, actually maybe this won't work as expected? Let's say you're _not_ on the Settings screen. In this case, the polyfills won't be registered and the result will be possibly that the required scripts are not registered, right?

I think the way to address this is to remove the logic here and move it to `is_needed`, so that the logic runs if:

We have the expected version of React _or_ we're on one of the AMP settings screens.

Reply:
> In this case, the polyfills won't be registered and the result will be possibly that the required scripts are not registered, right?

Since it has `'wp-api-fetch', 'wp-dom-ready', 'wp-i18n'` as dependency scripts while `wp_enqueue_script`, WP will enqueue them for it... right? I think WP prior to `5.6` already has these scripts' support and below `5.6` we are not exposing this service.

https://github.com/ampproject/amp-wp/blob/378522cfc969ed1d04abb1b0f1605838ac24f562/src/Admin/ValidationCounts.php#L78

Reply:
> We have the expected version of React or we're on one of the AMP settings screens.

`amp-validation-counts.js` doesn't require to react for its operation. Here is the dependecies for this:
```php
<?php return array('dependencies' => array('wp-api-fetch', 'wp-dom-ready', 'wp-i18n'), 'version' => '29b558177338ddf7e50f');
```